flytekit/core/data_persistence.py

```diff
@@ -558,6 +558,8 @@ def get_data(self, remote_path: str, local_path: str, is_multipart: bool = False
             pathlib.Path(local_path).parent.mkdir(parents=True, exist_ok=True)
             with timeit(f"Download data to local from {remote_path}"):
                 self.get(remote_path, to_path=local_path, recursive=is_multipart, **kwargs)
+        except FlyteDataNotFoundException:
```

Comment:
`FlyteDataNotFoundException` does not derive from `FlyteAssertion`, so any code that used to catch `FlyteAssertion` will no longer work. This is probably a reasonable change but worth calling out the risk

Reply:
yeah, I am not sure if we are catching `FlyteAssertion` anywhere besides that unit test. @pvditt do you have any opinion here?